crates/supervisor/storage/src/error.rs

```diff
@@ -62,4 +63,20 @@ impl PartialEq for StorageError {
     }
 }
 
+impl From<StorageError> for SuperchainDAError {
+    fn from(err: StorageError) -> Self {
+        match err {
+            StorageError::Database(_) => Self::DataCorruption,
+            StorageError::FutureData => Self::FutureData,
+            StorageError::EntryNotFound(_) => Self::MissedData,
+            StorageError::DatabaseNotInitialised => Self::UninitializedChainDatabase,
+            StorageError::ConflictError(_) => Self::ConflictingData,
+            StorageError::BlockOutOfOrder | StorageError::DerivedBlockOutOfOrder => {
+                Self::OutOfOrder
+            }
+            _ => err.into(),
```

Comment:
I am not comfortable with this. There are other error type defined which isn't mentioned in the matching. 
The `_ => err.into()` line will cause infinite recursion and a stack overflow at runtime if any unhandled variant is matched.